test/serializers/adapter_for_test.rb

```diff
@@ -138,14 +138,14 @@ def test_inherited_adapter_hooks_register_adapter
         Object.send(:remove_const, :MyAdapter)
       end
 
-      def test_inherited_adapter_hooks_register_demodulized_adapter
+      def test_inherited_adapter_hooks_register_namespaced_adapter
         Object.const_set(:MyNamespace, Module.new)
         MyNamespace.const_set(:MyAdapter, Class.new)
         my_adapter = MyNamespace::MyAdapter
         ActiveModel::Serializer::Adapter.inherited(my_adapter)
-        assert_equal ActiveModel::Serializer::Adapter.lookup(:my_adapter), my_adapter
+        assert_equal ActiveModel::Serializer::Adapter.lookup(:'my_namespace/my_adapter'), my_adapter
```

Comment:
Nice, I think this resolves an issue someone brought up earlier in the week User::SomethingSerializer